simplex-messaging-implementation.md

```diff
@@ -214,86 +191,103 @@ All further requests to the recipient and sender connection URIs MUST be rejecte
+
+__Please note__: server implementations MUST NOT track in any form how many times or whether the messages were retrieved.
+
+If the unknown message ID is passed in `message_id` parameter, the request should be rejected with HTTP status code 400 (Bad Request).
+
+If the request is successful, the server MUST respond with HTTP status code (200) returning as response body the requested message with the following properties:
+- `id` (string): server-generated unique ID allowing to identify messages until (they are deleted from the server) and to paginate responses.
+- `ts` (string): server timestamp of the time when the message was received from the sender.
+- `size` (number): message size, in bytes.
+- `msg` (string): encrypted message body, that the recipient should be able to decrypt with the key `EK`.
 
 
 #### Delete messages
 
-URI: `<RU>/messages`
+DELETE: `<RU>/messages`
 
 Example: DELETE `https://example.com/connection/aZ9f/messages`
 
-To delete messages from the connection, simplex messaging client MUST send DELETE request to the recipient connection URI `RU` (returned by the server when creating the connection) with the REQUIRED appended string `/messages` (it MUST NOT be changed by any implementation or deployment), signed with the key `RK`.
+To delete messages from the connection, simplex messaging client MUST send DELETE request to the recipient connection URI `RU` (returned by the server when creating the connection) with the REQUIRED appended string `/messages` (it MUST NOT be changed by any implementation or deployment). The request MUST be signed with the key `RK`.
 
-This request SHOULD be used by simplex messaging clients to delete the previously retrived messages when `"keepMessages": true` parameter was used or in case they no longer require to retrive the messages.
+Simplex messaging clients MUST use this endpoint to delete the previously retrived and stored (or processed) messages.
 
 Request body should be sent as JSON object with the following properties:
-- `connectionID` (string): existing connection ID (see [Connection ID](#connection-id)).
-- `pageSize` (number, optional): if set, the server will delete up to the requested number of messages, otherwise all messages, in both cases from the message ID in `fromMessageID` parameter (or from the earliest available).
-- `fromMessageID` (string, optional): the server will delete the messages received, starting from the message with the server message ID passed in this parameter. If not specified, it defaults to the server message ID of the earliest received message.
+- `fromMessageID` (string): the server will delete the messages starting from the message with the server message ID passed in this parameter.
+- `count` (number): the server will delete up to the requested number of messages from the message ID in `fromMessageID` parameter.
```

Comment:
why not make `count` optional to make client slightly more simple? e.g. when there'no large messages and it's the only/the last page it could just request to delete all messages starting with `fromMessageID`. Also wouldn't `upToMessageID` be simpler than `count`? the client workflow to request to message deletion would already be somewhat convoluted with necessity to keep large messages, I think it might be easier to skip them with [`fromMessageID`, `upToMessageID_1`), [`upToMessageID_1[+1]`, `upToMessageID_2`), etc. than with counts.

Reply:
proposed only single message deletion endpoint, as clients won't use bulk deletion normally and use websockets instead.